### PR TITLE
Add platform question to nps

### PR DESCRIPTION
**Pull request by gustavomm19**
https://github.com/breatheco-de/breatheco-de/issues/7574

### CODE REVIEW

**Review by tommygonzaleza — DISMISSED** [review]

File: breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py [hunk 1]
```diff
@@ -212,6 +239,10 @@ def test_role_assistant(self):
             model = self.bc.database.create(cohort=1, user=1, survey=1, cohort_user=cohort_users)
 
             generate_user_cohort_survey_answers(model.user, model.survey, status="OPENED")
+            for dict in self.bc.database.list_of("feedback.Answer"):
+                if dict["platform_and_content"]:
+                    print("dict")
+                    print(dict)
```
Comment (jefer94) [inline breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py:245]: 👀

File: breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py [hunk 2]
```diff
@@ -267,12 +298,25 @@ def test_role_assistant(self):
                             "user_id": n + 1,
                             "survey_id": n + 1,
                             "lowest": "not likely",
-                            "id": (n * 4) + 4,
+                            "id": (n * 5) + 4,
```
Comment (jefer94) [inline breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py:301]: why this is happening?

File: breathecode/feedback/models.py [hunk 3]
```diff
@@ -99,6 +99,7 @@ def __init__(self, *args, **kwargs):
     highest = models.CharField(max_length=50, default="very likely")
     lang = models.CharField(max_length=3, blank=True, default="en")
 
+    platform_and_content = models.BooleanField(default=False)
```
Comment (jefer94) [inline breathecode/feedback/models.py:102]: This solution is a bit inaccurate https://github.com/breatheco-de/breatheco-de/issues/7574#issuecomment-2249049862

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py [hunk 4]
```diff
@@ -118,12 +119,25 @@ def test_when_teacher_is_assigned(self):
...
                             "survey_id": n + 1,
                             "lowest": "not likely",
-                            "id": (n * 3) + 3,
+                            "id": (n * 4) + 3,
                             "highest": "very likely",
                             "cohort_id": None,
                             "academy_id": n + 1,
                         }
                     ),
+                    answer(
```
Comment (jefer94) [inline breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py:128]: There isn't any test for the case of a sass academy and cohort. two cases

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/feedback/models.py [hunk 5]
```diff
@@ -99,7 +102,9 @@ def __init__(self, *args, **kwargs):
     highest = models.CharField(max_length=50, default="very likely")
     lang = models.CharField(max_length=3, blank=True, default="en")
 
-    platform_and_content = models.BooleanField(default=False)
+    aditional_question = models.CharField(
```
Comment (jefer94) [inline breathecode/feedback/models.py:105]: There is a typo https://www.oxfordlearnersdictionaries.com/us/definition/english/additional?q=additional

Comment (jefer94) [reply breathecode/feedback/models.py:105]: `additional_question` isn't descriptive enough

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py [hunk 6]
```diff
@@ -77,53 +78,57 @@ def test_when_teacher_is_assigned(self):
...
+                    "cohort_id": None,
+                    "academy_id": n + 1,
+                },
+                {
+                    "title": f"How has been your experience with the platform and content?",
+                    "lowest": "not good",
+                    "highest": "very good",
+                    "cohort_id": None,
+                    "academy_id": None,
+                    "question_by_slug": "PLATFORM",
```
Comment (jefer94) [inline breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py:116]: If a cohort is set and is available as saas or an academy is not available as saas, only an answer should be generated related to `"question_by_slug": "PLATFORM",`

Comment (jefer94) [reply breathecode/feedback/tests/tasks/tests_generate_user_cohort_survey_answers.py:116]: So, I mean a case must be 

```py
answers = [
                {
                    "title": f"How has been your experience with the platform and content?",
                    "lowest": "not good",
                    "highest": "very good",
                    "cohort_id": None,
                    "academy_id": None,
                    "question_by_slug": "PLATFORM",
                 },
]
```

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/feedback/tasks.py [hunk 7]
```diff
@@ -110,31 +115,36 @@ def new_answer(answer: Answer):
...
-            cohort=survey.cohort, role="ASSISTANT", educational_status__in=["ACTIVE", "GRADUATED"]
-        )
-        count = 0
-        for ca in cohort_assistant:
-            if count >= survey.max_assistants_to_ask:
-                break
-            answer = Answer(mentor=ca.user, cohort=survey.cohort, academy=survey.cohort.academy, lang=survey.lang)
-            _answers.append(new_answer(answer))
-            count = count + 1
+        if not survey.cohort.available_as_saas:
```
Comment (jefer94) [inline breathecode/feedback/tasks.py:118]: `cohort.available_as_saas` could be None in this case it uses `cohort.academy.available_as_saas`

**Review by jefer94 — CHANGES_REQUESTED** [review]

File: breathecode/feedback/tasks.py (hunk 7 above)
Comment (jefer94) [reply breathecode/feedback/tasks.py:118]:
```py
        if survey.cohort.available_as_saas is False or (survey.cohort.available_as_saas is None and survey.cohort.academy.available_as_saas):
```

**Review by jefer94 — APPROVED** [review]

**Pull request merged**